### PR TITLE
Git Pre Push Hook

### PR DESCRIPTION
**Pull request by hghazzi**
### This PR will...

Trigger `eslint` and `karma` to run before any push occurs. You can still force a push by running git `push -n (or --no-verify)`, though I would avoid doing so.

**NOTE: `grunt hooks` needs to be run locally before this takes effect.**

### Why is this Pull Request needed?

To ensure we do not check in any code that could cause the build to fail.

### Are there any points in the code the reviewer needs to double check?

Would be great to bring down the runtime of both `eslint` and `karma` as much as possible. For `karma`, I'm only running unit tests on PhantomJS and not running istanbul. Maybe we can get some speedups from making changes like the ones @robwalch made in https://github.com/jwplayer/jwplayer-ads-googima/pull/241

### CODE REVIEW

File: .github/hooks/pre-push [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+echo "Running ESLint & Karma"
+
+grunt lint
```
Comment (robwalch) [inline .github/hooks/pre-push]: grunt is for human use. Script with underlying commands: run `eslint ./src/js` or the `npm run lint` if you want js and less linting.

**Comment by hghazzi** [comment]
@robwalch Got it, done

**Comment by hghazzi** [comment]
test this please

**Comment by hghazzi** [comment]
test this please

**Comment by hghazzi** [comment]
test this please

**Comment by hghazzi** [comment]
Test this please

File: README.md [hunk 2]
```diff
@@ -113,6 +113,12 @@ grunt karma:{BROWSER} e.g. grunt karma:chrome
 npm run lint
 ````
 
+6. Setup git pre-push hook
+# Will lint you code and run unit tests every time you push to remote
```
Comment (egreaves) [inline README.md]: your*

File: .github/hooks/pre-push [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+echo "Running ESLint & Karma"
+
+npm run lint
+lint=$?
+if [ $lint -ne 0 ]; then
+    echo "ESLint Failed. Use --no-verify to force push. Please don't. \n\n\n\n"
```
Comment (egreaves) [inline .github/hooks/pre-push]: Use `--no-verify` or `-n`? Also, if we don't want people to do this, should we even include it?

**Comment by hghazzi** [comment]
test this please

**Pull request merged**